net/sniproxy/files/sniproxy.init

```diff
@@ -0,0 +1,170 @@
+	echo "" >> "${RUNTIME_CONF_FILE}"
+}
+
+cfg_add_default_logs()
+{
+	local SECTION="${1}"
+	local RUNTIME_CONF_FILE="${2}"
+	local EXTRA_INDENT="${3}"
+
+	if [[ -z "${SECTION/cfg[0-9]*/}" ]]; then
+		cfg_add_logs ${SECTION} "${RUNTIME_CONF_FILE}"
+	fi
+}
+
+cfg_add_listeners()
+{
+	local SECTION="${1}"
+	local RUNTIME_CONF_FILE="${2}"
+	local LISTENER_NAME="${SECTION/cfg[0-9]*/}"
+
+	[[ -z "${LISTENER_NAME}" ]] || echo "# ${LISTENER_NAME}" >> "${RUNTIME_CONF_FILE}"
+
+	config_get ADDRESS ${SECTION} "address"
+	echo "listener ${ADDRESS} {" >> "${RUNTIME_CONF_FILE}"
+
+	config_get PROTOCOL ${SECTION} "protocol"
+	[[ -z "${PROTOCOL}" ]] || echo "	protocol ${PROTOCOL}" >> "${RUNTIME_CONF_FILE}"
+
+	config_get REUSEPORT ${SECTION} "reuseport"
+	[[ -z "${REUSEPORT}" ]] || echo "	reuseport ${REUSEPORT}" >> "${RUNTIME_CONF_FILE}"
+
+	config_get IPV6_V6ONLY ${SECTION} "ipv6_v6only"
+	[[ -z "${IPV6_V6ONLY}" ]] || echo "	ipv6_v6only ${IPV6_V6ONLY}" >> "${RUNTIME_CONF_FILE}"
+
+	config_get TABLE ${SECTION} "table"
+	[[ -z "${TABLE}" ]] || echo "	table ${TABLE}" >> "${RUNTIME_CONF_FILE}"
+
+	config_get FALLBACK ${SECTION} "fallback"
+	[[ -z "${FALLBACK}" ]] || echo "	fallback ${FALLBACK}" >> "${RUNTIME_CONF_FILE}"
+
+	config_get SOURCE ${SECTION} "source"
+	[[ -z "${SOURCE}" ]] || echo "	source ${SOURCE}" >> "${RUNTIME_CONF_FILE}"
+
+	config_get ACCESS_LOG ${SECTION} "access_log"
+	[[ -z "${ACCESS_LOG}" ]] || cfg_add_logs "${ACCESS_LOG}" "${RUNTIME_CONF_FILE}" "	"
+
+	config_get BAD_REQUESTS ${SECTION} "bad_requests"
+	[[ -z "${BAD_REQUESTS}" ]] || echo "	bad_requests ${BAD_REQUESTS}" >> "${RUNTIME_CONF_FILE}"
+
+	echo "}" >> "${RUNTIME_CONF_FILE}"
+	echo "" >> "${RUNTIME_CONF_FILE}"
+}
+
+cfg_add_table_entry()
+{
+	local ENTRY="${1}"
+	local RUNTIME_CONF_FILE="${2}"
+
+	[[ -z "${ENTRY}" ]] || echo "	${ENTRY}" >> "${RUNTIME_CONF_FILE}"
+}
+
+cfg_add_tables()
+{
+	local SECTION="${1}"
+	local RUNTIME_CONF_FILE="${2}"
+	local TABLE_NAME="${SECTION/cfg[0-9]*/}"
+
+	echo "table ${TABLE_NAME} {" >> "${RUNTIME_CONF_FILE}"
+
+	config_list_foreach ${SECTION} entry cfg_add_table_entry "${RUNTIME_CONF_FILE}"
+
+	echo "}" >> "${RUNTIME_CONF_FILE}"
+	echo "" >> "${RUNTIME_CONF_FILE}"
+}
+
+generate_conf() {
+	local CONF_FILE="${1}"
+	local RUNTIME_CONF_FILE="${2}"
+
+	config_load ${SERVICE_NAME}
+
+	echo "# Autogenerated from ${CONF_FILE}" > "${RUNTIME_CONF_FILE}"
+	echo "" >> "${RUNTIME_CONF_FILE}"
+
+	config_foreach cfg_add_globals "sniproxy" "${RUNTIME_CONF_FILE}"
+	config_foreach cfg_add_resolvers "resolver" "${RUNTIME_CONF_FILE}"
+	config_foreach cfg_add_default_logs "error_log" "${RUNTIME_CONF_FILE}"
+	config_foreach cfg_add_default_logs "access_log" "${RUNTIME_CONF_FILE}"
+	config_foreach cfg_add_listeners "listener" "${RUNTIME_CONF_FILE}"
+	config_foreach cfg_add_tables "table" "${RUNTIME_CONF_FILE}"
+}
+
+start_service() {
+	local CONF_FILE="/etc/config/${SERVICE_NAME}"
+	local RUNTIME_CONF_FILE="/etc/${SERVICE_NAME}.conf"
```

Comment:
I would suggest that we but the runtime config into a tmp folder to decrease the writes to the NAND.

Reply:
Just thought of this when I woke up, good catch!
Fixed now.